pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -27,12 +27,11 @@
 from pyomo.opt import TerminationCondition as tc
 from pyomo.core import Constraint, minimize, value, maximize
 cplex, cplex_available = attempt_import('cplex')
-from cplex.callbacks import LazyConstraintCallback
 
 logger = logging.getLogger('pyomo.contrib.mindtpy')
 
 
-class LazyOACallback_cplex(LazyConstraintCallback):
+class LazyOACallback_cplex(cplex.callbacks.LazyConstraintCallback):
```

Comment:
This won't help when cplex is not available, as `cplex.callbacks` will trigger the import of `cplex`.  Instead, try:
```python
class LazyOACallback_cplex(cplex.callbacks.LazyConstraintCallback if cplex_available else object):
```

Reply:
Thanks